content/en/docs/armory-agent/agent-mtls.md

```diff
@@ -0,0 +1,267 @@
+
+{{< prism lang="yaml" line="12-20" >}}
+apiVersion: spinnaker.armory.io/{{< param "operator-extended-crd-version" >}}
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      clouddriver:
+        # See https://docs.armory.io/docs/installation/armory-agent/agent-options/
+        kubesvc:
+          cluster: redis
+          grpc:
+            server:
+              security:
+                enabled: true
+                certificateChain: file:<path-to-agent.crt>
+                trustCertCollection: file:<path-to-agent.crt>
+                privateKey: file:<path-to-agent.key>
+                clientAuth: REQUIRE
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-plugin-options.md" >}} page for additional options.
+
+## Agent service configuration
+
+### Create a secret
+
+Create a secret in the target cluster namespace where the Agent resides.
+
+```bash
+kubectl create secret generic <agent-secret-name> \
+--from-file=agent.crt --from-file=agent.key
+```
+
+### Modify deployment configuration
+
+Modify the Agent's deployment configuration to mount the certs. @TODO what file is this in?
+
+If you have a custom CA, you need to mount the cert into the known trusted cert location:  `/etc/ssl/cert.pem`. Golang apps in Alpine use `/etc/ssl/cert.pem` as the source of a trusted CA.
+
+  1. Obtain a copy of the `cert.pem` file from a running Agent.
+  2. Append your custom CA onto the copied `cert.pem`  (for example `cat ca.crt >> cert.pem`)
+  3. Mount `cert.pem` to Agent deployment
+
+@TODO is this config correct? do the "this didn't work" sections work now or not?
+
+{{< prism lang="yaml"  >}}
+spec:
+  template:
+    spec:
+      containers:
+        volumeMounts:
+        - mountPath: /opt/kubesvc/cert
+          name: armoryagentcert
+        - mountPath: /opt/spinnaker/config
+          name: volume-kubesvc-config
+        - mountPath: /opt/kubesvc/cacert  # this didn't work as of Nov 2020
+          name: clouddrivercacert         # this didn't work as of Nov 2020
+      volumes:
+      - name: armoryagentcert
+        secret:
+          secretName: <agent-secret-name>
+      - name: clouddrivercacert         # this didn't work as of Nov 2020
+        secret:                         # this didn't work as of Nov 2020
+          secretName: <your-secret-name> # this didn't work as of Nov 2020
+      - name: certpem
+        secret:
+          secretName: ??? @TODO what secret is this?
+{{< /prism >}}
+
+
+### Configure the service
+
+@TODO what file is this in?
+
+{{< prism lang="yaml" >}}
+clouddriver:
+  grpc: <:443
+  insecure: false
+  tls:
+    #serverName: my-ca  #to override the server name to verify
+    insecureSkipVerify: false #if true, don't verify server's cert
+    clientKeyFile: /opt/kubesvc/cert/agent.key #ref to the private key (mTLS)
+    #clientKeyPassword: #if the clientKeyFile is password protected
+    #cacertFile: /opt/kubesvc/cacert/ca.crt #to validate server's cert
```

Comment:
I think the comments in this file section (190-194) should be completely removed. Instead, for explanation there should be a link to the agent docs section for these settings: https://docs.armory.io/docs/armory-agent/agent-options/

I also recommend removing the `cacertFile` field since the above documentation recommended appending the system cert file, which is the one that is used by default.